version: '3'

services:
  web:
    build: .
    command: bash -c "python manage.py makemigrations && python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/code
      - ./sample_datasets:/code/sample_datasets
      - ./simulator_api/migrations:/code/simulator_api/migrations
    ports:
      - "8000:8000"
    environment:
      - POSTGRES_NAME=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    depends_on:
      - db
    networks:
      - nifi-network
  db:
    image: postgres
    environment:
      - POSTGRES_PASSWORD=postgres
    ports:
      - "54320:5432" #Port 5432 was already used by some other process
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - nifi-network

  zookeeper:
    image: confluentinc/cp-zookeeper:6.1.0
    hostname: zookeeper
    ports:
        - "2181:2181"
    environment:
        ZOOKEEPER_CLIENT_PORT: 2181
        ZOOKEEPER_TICK_TIME: 2000
        ZOOKEEPER_SYNC_LIMIT: 2
    networks:
        - nifi-network

  kafka:
    image: confluentinc/cp-kafka:6.1.0
    hostname: kafka
    ports:
      - "9092:9092"
    depends_on:
      - zookeeper
    environment:
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092
      KAFKA_BROKER_ID: 1
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_ZOOKEEPER_CONNECT: "zookeeper:2181"
      KAFKA_AUTO_CREATE_TOPICS_ENABLE: "true"
      KAFKA_DELETE_TOPIC_ENABLE: "true"
    networks:
      - nifi-network

volumes:
  postgres_data:

networks:
  nifi-network:
    external: true



